server/game/cards/locations/04/harrenhalffh.js

```diff
@@ -1,38 +1,36 @@
 const DrawCard = require('../../../drawcard.js');
 
 class Harrenhal extends DrawCard {
-    setupCardAbilities() {
+    setupCardAbilities(ability) {
         this.action({
             title: 'Kneel Harrenhal to put a character in play',
-            method: 'kneel',
-            phase: 'challenge'
+            cost: ability.costs.kneelSelf(),
+            phase: 'challenge',
+            handler: context => {
```

Comment:
I think in this case it's because the onSelect for the prompt hasn't been separated into its own method. Otherwise, this is literally the same level of indentation as reactions / interrupts.

I plan on pulling most (thought probably not 100%) of the select prompts into another action/reaction/interrupt property, which will further collapse things like this, but have been busy working on other things.